app/src/main/java/openfoodfacts/github/scrachx/openfood/features/scan/ContinuousScanActivity.kt

```diff
@@ -804,31 +721,26 @@ class ContinuousScanActivity : BaseActivity(), IProductView {
         }
     }
 
-    private inner class BarcodeScannerCallback : BarcodeCallback {
-        override fun barcodeResult(result: BarcodeResult) {
-            hintBarcodeDisp?.cancel()
-
-            // Prevent duplicate scans
-            if (result.text == null || result.text.isEmpty() || result.text == lastBarcode) return
+    private fun onBarcodeResult(text: String?) {
+        hintBarcodeDisp?.cancel()
 
-            val invalidBarcode = daoSession.invalidBarcodeDao.unique {
-                where(InvalidBarcodeDao.Properties.Barcode.eq(result.text))
-            }
-            // Scanned barcode is in the list of invalid barcodes, do nothing
-            if (invalidBarcode != null) return
+        // Prevent duplicate scans
+        if (text == null || text.isEmpty() || text == lastBarcode) return
 
-            if (beepActive) {
-                beepManager.playBeepSound()
-            }
-            lastBarcode = result.text
-            if (!isFinishing) {
-                setShownProduct(result.text)
-                matomoAnalytics.trackEvent(AnalyticsEvent.ScannedBarcode(result.text))
-            }
+        val invalidBarcode = daoSession.invalidBarcodeDao.unique {
+            where(InvalidBarcodeDao.Properties.Barcode.eq(text))
         }
+        // Scanned barcode is in the list of invalid barcodes, do nothing
+        if (invalidBarcode != null) return
```

Comment:
Missing brackets here :)

Reply:
I usually don't put brackets if I only return. We should set a checkstyle here :)